### ISSUE TITLE
App crashes once at a song page on UG

### ISSUE THREAD

**Issue by terryt063** [thebigg73/OpenSongTablet#261]
Running v6.1.1

Add song, from online, enter search term, ultimate guitar opens, select song, Centre button doesn't swap from magnifying glass to a plus, 2-3 secs later, app crashes.

Only updated it recently, but has previously been awesome for new songs....

**Comment by thebigg73** [comment]
Hi,
Can you locate the OpenSong/Settings/CrashLog.txt file and post it here please

**Comment by terryt063** [comment]
```
java.lang.IndexOutOfBoundsException: Inconsistency detected. Invalid item position 353(offset:353).state:366 androidx.recyclerview.widget.RecyclerView{b6afa7b VFED..C.. ......ID 0,0-1008,1670 #7f0a0798 app:id/songListRecyclerView}, adapter:com.garethevans.church.opensongtablet.songmenu.SongListAdapter@aca5941, layout:com.garethevans.church.opensongtablet.songmenu.SongMenuFragment$1@b005098, context:com.garethevans.church.opensongtablet.MainActivity@48708c0
	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6821)
	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6757)
	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6753)
	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2362)
	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1662)
	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4645)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4348)
	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4919)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
	at android.widget.LinearLayout.layoutHorizontal(LinearLayout.java:1880)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1640)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:919)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at androidx.recyclerview.widget.RecyclerView$LayoutManager.layoutDecoratedWithMargins(RecyclerView.java:10407)
	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1720)
	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4645)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4348)
	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4919)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at androidx.viewpager2.widget.ViewPager2.onLayout(ViewPager2.java:535)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1287)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1891)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1729)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1638)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at com.android.internal.policy.DecorView.onLayout(DecorView.java:1094)
	at android.view.View.layout(View.java:25754)
	at android.view.ViewGroup.layout(ViewGroup.java:6822)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:5236)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:4502)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:3275)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:11257)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1650)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659)
	at android.view.Choreographer.doCallbacks(Choreographer.java:1129)
	at android.view.Choreographer.doFrame(Choreographer.java:1055)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:230)
	at android.os.Looper.loop(Looper.java:319)
	at android.app.ActivityThread.main(ActivityThread.java:8913)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```

**Comment by thebigg73** [comment]
Thanks for sending this

Grrr.  I thought I'd fixed this bug!
This is caused by the song menu list not being what the adapter is expecting (i.e. it has the wrong number of items in it when it tries to query it) - this is the inconsistency.  Does this happen all of the time when you repeat the above action?  I can't get it to crash on my devices, so I'm struggling to find out exactly why this happens

**Comment by terryt063** [comment]
Yep, at least it was this morning. 5 attempts got 5 crashes and after the second was trying different songs.

I've got a couple other android devices (that was a Samsung Tab S6Lite) and see if its just that one.

**Comment by magicvince** [comment]
Have the same issue, and I'll be volunteer to test an apk with correction :)

**Comment by thebigg73** [comment]
I've pushed a potential fix to GitHub just now.  If you use Android studio and are happy with testing that, then I'd appreciate this!
If you're not too sure about how to do this and would rather an apk file, I've uploaded a test.apk file to my downloads page (you'll likely need to uninstall your existing version)

**Comment by magicvince** [comment]
I'm tried the test.apk, and always no download icon when I display the song in opensong with desktop view.

**Comment by thebigg73** [comment]
> I'm tried the test.apk, and always no download icon when I display the song in opensong with desktop view.

Not quite sure what you mean with 'desktop view'?
Is this issue trying to download the apk file?  Does the app crash?

**Comment by terryt063** [comment]
So it doesn't crash, but the magnifying glass doesn't turn to a plus... Can't import...

Guessing desktop view is meant by the option to report as a desktop browser instead of a mobile browser. Them app pop-ups are quite the nuisance

![Screenshot_20240420_171021_OpenSongApp](https://github.com/thebigg73/OpenSongTablet/assets/59738171/e476e60b-7601-44ee-b6f0-a8d08917e987)

**Comment by thebigg73** [comment]
The UG issue is unrelated and due to a change in their website (I have, however, made a fix for this based on your last message).

I'll reupload a new test.apk..

**Comment by thebigg73** [comment]
New test.apk uploaded - it should show as v6.1.2 (sneak release with some other fixes and hopefully what goes live on Monday)

**Comment by thebigg73** [comment]
Can you confirm if this works and (more importantly) that the crash no longer happens for you

**Comment by terryt063** [comment]
> Can you confirm if this works and (more importantly) that the crash no longer happens for you

Got to run out the door for a gig, but will give it a spin when I can...

**Comment by terryt063** [comment]
The crashing has gone though on the previous apk. Tested on the Samsung Tab S6Lite and a really old Nexus 7 I had laying about (both crashed with the Google play version)

**Comment by thebigg73** [comment]
Have a great gig and hopefully you're able to use the non-crashing version!!!

**Comment by magicvince** [comment]
Seems to work fine for me too. (Motorola G7+ with LineageOS 20)
Thanks for your reactivity

**Comment by thebigg73** [comment]
Fix released in v6.1.2

**Comment by terryt063** [comment]
Morning! Thank you kindly for this. Tested this morning and it functions as it should. Only odd behaviour seen is that it now can try to import the UG listing page (thought that didn't happen previously...)

It doesn't save though and just a "well, don't do silly things" but thought I'd explore.

The crash log gives this.

```
Please share this with crashlog@opensongapp.com
=============================
OpenSongApp version: V6.1.2
Android API: 34
Device: gta4xlvewifi
Crash time: Apr 21, 2024 11:15:38
=============================
If possible, please give a brief description of what you were doing when the app crashed:





=============================
java.lang.IllegalArgumentException: Wrong state class, expecting View State but received class com.google.android.material.stateful.ExtendableSavedState instead. This usually happens when two views of different type have the same id in the same hierarchy. This view's id is id/0x27. Make sure other views do not use the same id.
	at android.view.View.onRestoreInstanceState(View.java:23540)
	at android.widget.TextView.onRestoreInstanceState(TextView.java:7129)
	at android.view.View.dispatchRestoreInstanceState(View.java:23512)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at com.google.android.material.textfield.TextInputLayout.dispatchRestoreInstanceState(TextInputLayout.java:3185)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:4253)
	at android.view.View.restoreHierarchyState(View.java:23490)
	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:698)
	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:3176)
	at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:3161)
	at androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:619)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:275)
	at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:114)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1455)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3034)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2952)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:263)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:350)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:251)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1582)
	at android.app.Activity.performStart(Activity.java:8993)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:4198)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:225)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:205)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:177)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:230)
	at android.os.Looper.loop(Looper.java:319)
	at android.app.ActivityThread.main(ActivityThread.java:8913)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```
